### PR TITLE
Andrew

### PR DESCRIPTION
**Pull request by AndrewKoch**

### CODE REVIEW

File: heathskin/game_state.py [hunk 1]
```diff
@@ -18,34 +21,39 @@ def build_from_entities(cls, entities):
         gs.entities = entities
         return gs
 
-    def __init__(self):
+    def __init__(self, replay_from_log=False):
```
Comment (rhigdon) [inline heathskin/game_state.py]: rename to `write_db_history` instead of `replay_from_log`

File: heathskin/game_state.py [hunk 2]
```diff
@@ -78,6 +86,13 @@ def _get_heroes_from_entities(self):
     def _is_player_first(self):
         return self.entities.get('36').get_tag('ZONE') == 'OPPOSING PLAY (Hero)'
 
+    def _get_player_healths(self):
+        player1_hero, player2_hero = self._get_heroes_from_entities()
+        player1_health = (30 - player1_hero.get_tag('DAMAGE', 0))
```
Comment (rhigdon) [inline heathskin/game_state.py:92]: what happens with armor? what happens with cards that grant bonuses to health?

File: heathskin/game_state.py [hunk 3]
```diff
@@ -99,13 +114,17 @@ def feed_line(self, line):
 
             self.parser.feed_line(**results.groupdict())
 
-        if self.is_gameover():
+        if self.is_gameover() and not self.replay_from_log:
             card_db = card_database.CardDatabase.get_database()
             our_hero, enemy_hero = self._get_heroes_from_entities()
+            our_health, enemy_health = self._get_player_healths()
```
Comment (rhigdon) [inline heathskin/game_state.py]: move this into `self._create_history()`

File: heathskin/game_state.py [hunk 4]
```diff
@@ -99,13 +114,17 @@ def feed_line(self, line):
...
+
             self._create_history(**{
                 'hero': card_db.get_card_by_id(our_hero.card_id)['name'],
                 'opponent': card_db.get_card_by_id(enemy_hero.card_id)['name'],
                 'turns': self.entities.get('1').get_tag('TURN'),
+                'hero_heath': our_health,
```
Comment (rhigdon) [inline heathskin/game_state.py]: delete these lines after moving the above

File: heathskin/log_parser.py [hunk 5]
```diff
@@ -326,6 +331,13 @@ def _tag_change(self, msg):
             # raise PreventableException("failed to handle tag change for msg: '{}'".format(msg))
 
         target_ent.update_tag(**results)
+        if results.get('tag_name') == 'DAMAGE':
```
Comment (rhigdon) [inline heathskin/log_parser.py]: remove this logging?

File: heathskin/game_state.py (hunk 2 above)
Comment (AndrewKoch) [inline heathskin/game_state.py:92]: As is, this correctly tracks the hero health, factoring in armor and healing. In the future, I would like to implement these stats separately.

File: heathskin/game_state.py [hunk 6]
```diff
@@ -18,40 +21,48 @@ def build_from_entities(cls, entities):
...
+
+        self.start_new_game()
+
         self.game_type = None
 
     def _create_history(self, *args, **kwargs):
         """ Create Game History after game Ends
         """
         history = GameHistory()
+        history.start_time = datetime.now()
```
Comment (rhigdon) [inline heathskin/game_state.py]: this is wrong: should be set on the GameEntity object during `start_game()`

File: heathskin/game_state.py [hunk 7]
```diff
@@ -18,40 +21,48 @@ def build_from_entities(cls, entities):
...
-          else:
-            history.player2 = player.get('username')
-        db.session.add(history)
-        db.session.commit()
+            if player.get('first'):
+                history.player1 = player.get('username')
+            else:
+                history.player2 = player.get('username')
+
+        if not self.replay_from_log:
```
Comment (rhigdon) [inline heathskin/game_state.py]: move this back inside game_state: don't even call `_create_history` if `self.replay_from_log` is set

File: heathskin/game_state.py [hunk 8]
```diff
@@ -129,6 +148,9 @@ def start_new_game(self):
         self.parser = LogParser(self)
         self.players = {}
 
+        if self.replay_from_log:
```
Comment (rhigdon) [inline heathskin/game_state.py]: this is dead code now... intended?

**Pull request merged**